gitpod-io/gitpod-vscode-desktop
# Add Heartbeat

Move heartbeat to gitpod-desktop from gitpod-remote

How to test:
- [Run this extension](https://github.com/gitpod-io/gitpod-vscode-desktop/blob/master/docs/CONTRIBUTING.md)
- Connect to a workspace in this prev env (use SSH gateway for now) https://jp-heartbeat-test.preview.gitpod-dev.com/workspaces (https://github.com/gitpod-io/gitpod/pull/11021)
    - make sure you have the debug VSCode Desktop window focus
- Install this gitpod-remote vsix [gitpod-remote-ssh-0.0.42.zip](https://github.com/gitpod-io/gitpod-vscode-desktop/files/9089328/gitpod-remote-ssh-0.0.42.zip) (related PR https://github.com/gitpod-io/openvscode-server/pull/378)
    - Upload `.zip` file to workspace, unzip it
    - Right-click on `.vsix` file and install it
- Check `__gitpod.cancelGitpodRemoteHeartbeat command executed` is logged in the gitpod-remote logs
- Open Gitpod with this PR https://github.com/gitpod-io/openvscode-server/pull/378
- Exec commands below to see if heartbeat sent
```
cd dev/gpctl
go build
./gpctl workspaces last-heartbeat <instance_id>
```
- Check only one close heartbeat is send

~~Not sure how to trigger ssh reconnection~~ :thinking: 

Maybe use `sudo killall sshd`, but window closed after sshd terminate, not sure if it works
